sqlmesh/core/plan/definition.py

```diff
@@ -700,6 +714,13 @@ def _refresh_dag_and_ignored_snapshots(self) -> None:
                 "Revert or apply changes before proceeding with restatements."
             )
 
+        if self._input_backfill_models is not None:
+            if not self.is_dev:
```

Comment:
do we need to be so strict?

Reply:
I strongly believe so. Why would anyone want to backfill prod partially?

Reply:
you never know

Reply:
I agree with being strict here unless we hear a good use case. The product is opinionated about characteristics that make up prod and we so far this seems like a good bet.